perceval/backends/stackexchange.py

```diff
@@ -0,0 +1,296 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2016 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+#
+# Authors:
+#   Alberto Martín <alberto.martin@bitergia.com>
+#
+
+import json
+import logging
+import os.path
+
+import requests
+import time
+
+from ..backend import Backend, BackendCommand, metadata
+from ..cache import Cache
+from ..errors import CacheError
+from ..utils import str_to_datetime, DEFAULT_DATETIME, urljoin
+
+# Filters are immutable and non-expiring. This filter allows to retrieve all
+# the information regarding Each question. To know more, visit
+# https://api.stackexchange.com/docs/questions and paste the filter in the
+# whitebox filter. It will display a list of checkboxes with the selected
+# values for the filter provided.
+
+QUESTIONS_FILTER = 'Bf*y*ByQD_upZqozgU6lXL_62USGOoV3)MFNgiHqHpmO_Y-jHR'
+MAX_QUESTIONS = 100  # Maximum number of reviews per query
+STACKEXCHANGE_API_URL = 'https://api.stackexchange.com'
+VERSION_API = '2.2'
+
+logger = logging.getLogger(__name__)
+
+
+def get_update_time(item):
+    """Extracts the update time from a StackExchange item"""
+    return item['last_activity_date']
+
+
+class StackExchange(Backend):
+    """StackExchange backend for Perceval.
+
+    This class retrieves the questions stored in any of the
+    StackExchange sites. To initialize this class the
+    site and the tag must be provided.
+
+    :param site: StackExchange site
+    :param tagged: filter items by question Tag
+    :param token: StackExchange access_token for the API
+    :param cache: cache object to store raw data
+    """
+    version = '0.1.0'
+
+    def __init__(self, site=None, tagged=None, token=None,
```

Comment:
According to the doc from above, `site` and `tagged` are mandatory. Then, why do they have a default value?. I guess that `site` is mandatory, but not `tagged` because maybe I want to have all the items without regard to which tag they have.